src/main/js/field/EndpointButtonField.js

```diff
@@ -43,13 +43,8 @@ class EndpointButtonField extends Component {
         const {
             fieldKey, csrfToken, onChange, currentConfig, endpoint, requestedDataFieldKeys
         } = this.props;
-        let newFieldModel = FieldModelUtilities.createEmptyFieldModel(requestedDataFieldKeys, currentConfig.context, currentConfig.descriptorName);
-        requestedDataFieldKeys.forEach((field) => {
-            const values = FieldModelUtilities.getFieldModelValues(currentConfig, field);
-            newFieldModel = FieldModelUtilities.updateFieldModelValues(newFieldModel, field, values);
-        });
+        const newFieldModel = FieldModelUtilities.createFieldModelFromRequestedFields(currentConfig, requestedDataFieldKeys);
```

Comment:
This change can probably be implemented in the other endpoint classes as well (TableSelectInput.js and EndpointSelectField.js).

Reply:
I will make those changes before merging.